python/cudf/cudf/tests/test_orc.py

```diff
@@ -1714,3 +1714,23 @@ def test_empty_columns():
 
     got_df = cudf.read_orc(buffer)
     assert_eq(expected, got_df)
+
+
+def test_orc_reader_zstd_compression(list_struct_buff):
+    try:
+        expected = cudf.read_orc(list_struct_buff)
+
+        # save with ZSTD compression
+        buffer = BytesIO()
+        pyarrow_tbl = pyarrow.orc.ORCFile(list_struct_buff).read()
+        writer = pyarrow.orc.ORCWriter(buffer, compression="zstd")
+        writer.write(pyarrow_tbl)
+        writer.close()
+
+        got = cudf.read_orc(buffer)
+        assert_eq(expected, got)
+    except RuntimeError as e:
+        if "Unsupported compression type" in str(e):
+            pytest.mark.xfail(reason="nvcomp build doesn't have zstd")
+        else:
+            raise e
```

Comment:
Since there is a lot happening in the try block, we can minimize the code to catch the error just from the right statement this way:
```suggestion
    expected = cudf.read_orc(list_struct_buff)
    # save with ZSTD compression
    buffer = BytesIO()
    pyarrow_tbl = pyarrow.orc.ORCFile(list_struct_buff).read()
    writer = pyarrow.orc.ORCWriter(buffer, compression="zstd")
    writer.write(pyarrow_tbl)
    writer.close()
    try:
        got = cudf.read_orc(buffer)
    except RuntimeError as e:
        if "Unsupported compression type" in str(e):
            pytest.mark.xfail(reason="nvcomp build doesn't have zstd")
        else:
            raise e
     assert_eq(expected, got)
```